### ISSUE TITLE
With long file names the tree letters do not show

### ISSUE THREAD

**Issue by maxbruening** [paul-schaaf/talon-filetree#27]
With long file names the tree letters do not show. Maybe it is possible to improve this? E.g. by always breaking the file names early enough so that there is space for the tree letters?

**Comment by david-tejada** [comment]
I have run into this issue too.

I prefer the second solution proposed by Paul. For the other solution we would need to check how many ancestors the tree item has to check where we should cut off the label. Which is doable, but we might run into the issue that we have no characters to display the labels of deeply nested tree items. Also, the user making the bar wider wouldn't reveal the full name of the folder/file, which could be frustrating.

I think the solution of placing the letters in front of the file name is overall better and I think it's easier to implement too. We would need to make the letters part of the label, since we can't move the position of the description to the front.

**Comment by paul-schaaf** [comment]
> For the other solution we would need to check how many ancestors the tree item has to check where we should cut off the label

Im not quite following this. How is the file name of a directory related to its children? In my head I do something like e.g.
```
const fileName = uri.length > settings.uriLength ? shorten(uri,settings.uriLength) : uri;
```
and that would be all

**Comment by david-tejada** [comment]
Take a look at this example. The label of the file `filename.txt` is cut off while the one for `Folder with a long name` is not, just because the first one is nested much deeper. If you were to have a setting of say 12, the letter for `filename.txt` still wouldn't show while *Folder with a long name* would be cut off unnecessarily.

<img width="208" alt="Screenshot 2023-09-25 at 18 31 15" src="https://github.com/paul-schaaf/talon-filetree/assets/3219671/d5c173d2-07f0-4357-9761-5ea01b2981de">

**Comment by maxbruening** [comment]
To me it seems that the solution of putting tree letters first is more practical. However, from a user experience, I am somewhat uncertain. It may be slightly more disorienting for users to have the letters first. But maybe this is not at all the case and having the letters first actually feels more natural.

**Comment by david-tejada** [comment]
I have done a quick implementation of both alternatives and here it's how it would look like:

Hints to the left with different separators:

<img width="437" alt="Screenshot 2023-09-30 at 12 44 41" src="https://github.com/paul-schaaf/talon-filetree/assets/3219671/74b0a9bc-76d4-427c-acf2-b497076a157c">
<img width="433" alt="Screenshot 2023-09-30 at 12 45 56" src="https://github.com/paul-schaaf/talon-filetree/assets/3219671/64b2d420-dc20-4985-90bd-cd53546556ba">
<img width="432" alt="Screenshot 2023-09-30 at 12 46 51" src="https://github.com/paul-schaaf/talon-filetree/assets/3219671/e28a5878-4b15-4fc1-9b98-4fdc1c533703">

Cutting off labels:

<img width="371" alt="Screenshot 2023-09-30 at 14 04 41" src="https://github.com/paul-schaaf/talon-filetree/assets/3219671/8bdc511d-d67d-4d05-9ecd-0733d6bce291">
<img width="370" alt="Screenshot 2023-09-30 at 14 06 51" src="https://github.com/paul-schaaf/talon-filetree/assets/3219671/13712681-8aa4-49b4-962f-8c754613e426">


For every level I have added four characters. It's not exact because the font is not monospace. We would also need to consider the indent of the tree as it can be changed by the user. We can look at the setting `workbench.tree.indent` for that and adjust accordingly. Overall I think it doesn't look bad and I favor this solution now. What do you guys think?

**Comment by paul-schaaf** [comment]
>Take a look at this example. The label of the file filename.txt is cut off while the one for Folder with a long name is not, just because the first one is nested much deeper. If you were to have a setting of say 12, the letter for filename.txt still wouldn't show while Folder with a long name would be cut off unnecessarily.

ah got it thanks!

personally I like adding the characters in the front (I like both `[<hint>]` and ` | `). But I like it just generally regardless of name lengths and want to try using it. We could still have the cutoff approach too tho if you prefer that. so i would suggest adding them both

**Comment by david-tejada** [comment]
Yeah, the good thing about having the hint in front is that its place is predictable. Easier to scan. I might go for that too.

Let's do one thing. Since that option is easier to implement I will create a PR for it. After using it for a while we can decide if we want the other option too.

**Comment by paul-schaaf** [comment]
sounds good!

**Comment by paul-schaaf** [comment]
closed for now by #29
